custom_components/ha-glocaltokens/sensor.py

```diff
@@ -21,25 +29,60 @@ class GlocaltokensSensor(GlocaltokensEntity):
     def __init__(self, coordinator, entry, device):
         """Initialize the sensor."""
         super().__init__(coordinator, entry)
-        self._name = device[GLOCALTOKENS_DEVICE_NAME]
-        self._token = device[GLOCALTOKENS_TOKEN]
+        self._name = device[DEVICE_NAME]
+        self._state = device[TOKEN]
 
     @property
-    def name(self):
-        """Return the name of the sensor."""
-        return f"{DEFAULT_NAME} {self._name}"
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        return {
+            "device": str(self.name),
+            "integration": DOMAIN,
+        }
 
-    @property
-    def state(self):
-        """Return the state of the sensor."""
-        return self._token
+class GlocaltokensAlarmSensor(GlocalAlarmEntity):
+    """Representation of a Sensor."""
+
+    def __init__(self, coordinator, entry, device):
+        """Initialize the sensor."""
+        super().__init__(coordinator, entry)
+        self._name = device[DEVICE_NAME]
+        self._alarms = device[ALARMS]
+
+        if len(self._alarms) > 0:
+            self._state = convert_from_ms_to_s(self._alarms[0][FIRE_TIME])
+        else:
+            self._state = STATE_OFF
 
     @property
-    def icon(self):
-        """Return the icon of the sensor."""
-        return ICON
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        return {
+            'alarms': self._alarms,
+            "device": str(self.name),
+            "integration": DOMAIN,
+        }
+
+
+class GlocaltokensTimerSensor(GlocalTimersEntity):
+    """Representation of a Sensor."""
+
+    def __init__(self, coordinator, entry, device):
+        """Initialize the sensor."""
+        super().__init__(coordinator, entry)
+        self._name = device[DEVICE_NAME]
+        self._timers = device[TIMERS]
+
+        if len(self._timers) > 0:
+            self._state = STATE_ON
+        else:
+            self._state = STATE_OFF
```

Comment:
Also very similar question (as above) raises here. So if Google Home devices has more then one timer we say that this sensor is on?

Reply:
All timers are listed as attributes, but see my answer above, maybe same things apply to this